.circleci/config.yml

```diff
@@ -122,7 +131,11 @@ references:
 
   yarn-install: &yarn-install
     name: Install dependencies
-    command: yarn install --frozen-lockfile
+    command: |
+      source "$HOME/.nvm/nvm.sh"
+      nvm use
+      npm install -g yarn
```

Comment:
> It's strange that we need to install yarn like this. Before this PR, where did the yarn executable come from in CircleCI when running yarn-install?

In a nutshell: for each command, Circle essentially uses a brand new shell instance that doesn't necessarily carry over the defaults from a prior instance. You can prove this out by using the `update-node` task and immediately validating what version of node is being used in a subsequent command. Here's where I confirmed this in CI:

_Test: Verify Node Version Without NVM Refresh_: [link](https://app.circleci.com/pipelines/github/Automattic/wp-calypso/65090/workflows/9e6046fb-b424-4187-9b52-a5963213bf07/jobs/787705)

```
// command: update-node

#!/bin/bash --login
set +e
set +x
export NVM_DIR="$HOME/.nvm" && (
  git clone https://github.com/nvm-sh/nvm.git "$NVM_DIR"
  cd "$NVM_DIR"
  git checkout v0.35.3
) && \. "$NVM_DIR/nvm.sh" --no-use
nvm install
nvm alias default "$(cat .nvmrc)"
nvm use

// command: Verify Node Version

#!/bin/bash --login
nvm list
yarn --version
/bin/bash: nvm: command not found 
1.12.1

// command: Install dependencies
yarn install v1.12.1
 
error wp-calypso-monorepo@0.17.0: The engine "node" is incompatible with this module. Expected version "^v12.18.0". Got "11.0.0"
error wp-calypso-monorepo@0.17.0: The engine "yarn" is incompatible with this module. Expected version "^1.22.4". Got "1.12.1"
```

You can see that by default, the shell is using Node v11.0.0 and yarn 1.12.1.

_Test: Verify Node Version After NVM Refresh_: [link](https://app.circleci.com/pipelines/github/Automattic/wp-calypso/65106/workflows/f584e8f0-dfd8-426f-9c2b-6f0329f2ef69/jobs/787804)

```
// Command: install dependencies

#!/bin/bash --login 
source "$HOME/.nvm/nvm.sh" 
yarn install --frozen-lockfile 
 
yarn install v1.12.1
error wp-calypso-monorepo@0.17.0: The engine "yarn" is incompatible with this module. Expected version "^1.22.4". Got "1.12.1"
error Found incompatible module
```

Now Node version is correct, but yarn is still wrong.

This could be a bug specific to Circle's Mac installation of nvm, but the implication is that nvm has to be refreshed as necessary after the initial setup so all bases are covered. Interestingly enough, this is how it has been wired up in wp-desktop's existing CI as well (see [here](https://github.com/Automattic/wp-desktop/blob/develop/.circleci/config.yml#L409) and [here](https://github.com/Automattic/wp-desktop/blob/develop/.circleci/config.yml#L421)), so this issue must have been encountered before.